cmd/tlsrouter/e2e_test.go

```diff
@@ -192,16 +192,16 @@ func selfSignedCert(domains []string) (tls.Certificate, *x509.CertPool, error) {
 			Organization: []string{"Test Co"},
 			CommonName:   domains[0],
 		},
-		NotBefore:             time.Time{},
+		NotBefore:             time.Now().Add(-5 * time.Minute),
 		NotAfter:              time.Now().Add(60 * time.Minute),
 		IsCA:                  true,
 		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
 		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
 		BasicConstraintsValid: true,
-		DNSNames:              domains[1:],
+		DNSNames:              domains[:],
```

Comment:
Why this change? Doesn't this map to additional domains, where domains[0] is the primary CN in the cert?

Reply:
This is because CommonName has long been marked as deprecated and was always meant to be duplicated in the subject alternative names (along with any additional names). In Go 1.16 the unittests would fail when a single domain was passed to this func because it left the subject alt names completely empty and was only using the common name field.

Reply:
i.e.,
```
--- FAIL: TestRouting (0.69s)
    e2e_test.go:82: get "test.com" failed: dial TLS "127.0.0.1:52685" for "test.com": 
    x509: certificate relies on legacy Common Name field, use SANs or temporarily 
    enable Common Name matching with GODEBUG=x509ignoreCN=0
```

Reply:
TIL, thanks!